arrow_pd_parser/caster.py

```diff
@@ -68,20 +68,173 @@ def _convert_str_to_datetime_obj_series(
     return s_new
 
 
-def _default_str_bool_mapper(s: Union[str, int, float]) -> bool:
+def _default_str_bool_mapper(
+    s: Union[str, int, float], bool_errors="coerce"
+) -> Union[bool, float]:
+    """
+    Maps a given input to a boolean value based on predefined strings.
+
+    Parameters:
+    -----------
+    s : Union[str, int, float]
+        The input value to be mapped. Can be a string, integer, or float.
+        Common values include 'yes', 'no', 'true', 'false', '1', '0'.
+
+    bool_errors : str, optional, default="coerce"
+        Specifies the behavior when the input value cannot be mapped to a boolean.
+        - "coerce": returns np.nan for unmappable values.
+        - "raise": raises a ValueError for unmappable values.
+
+    Returns:
+    --------
+    Union[bool, float]
+        The mapped boolean value if the input is recognized, 
+        or np.nan if the input is unrecognized and bool_errors is set to "coerce".
+        If bool_errors is set to "raise", a ValueError is raised for unrecognized inputs.
+
+    Raises:
+    -------
+    ValueError
+        If bool_errors is "raise" and the input value cannot be mapped to a boolean.
+
+    Notes:
+    ------
+    - Function is case-insensitive, 
+    - Ignores leading and trailing whitespace in the input string.
+    - Empty strings and None are mapped to np.nan.
+    - If the input is NaN (as determined by pandas.isna()), function returns np.nan.
+
+    Examples:
+    ---------
+    >>> _default_str_bool_mapper("yes")
+    True
+    >>> _default_str_bool_mapper("no")
+    False
+    >>> _default_str_bool_mapper("True")
+    True
+    >>> _default_str_bool_mapper("0")
+    False
+    >>> _default_str_bool_mapper("")
+    nan
+    >>> _default_str_bool_mapper("maybe", bool_errors="coerce")
+    nan
+    >>> _default_str_bool_mapper("maybe", bool_errors="raise")
+    Traceback (most recent call last):
+        ...
+    ValueError: maybe
+    """
+
+    basic_map = {
+        "yes": True,
```

Comment:
while I'm inclined to agree on wanting a reasonably narrow set of values to consider as truthy or falsy, I think that's not quite relevant for what this function is doing. If I understand correctly, this will function will be invoked when you have a column read as a string (e.g. from CSV) that metadata says is a boolean column. Hence, you would expect the column to be boolean-esque anyway.

There are certainly cases where the data is boolean-ish, but the source database metadata calls them strings (caseman has a couple of flag columns that are just `Y`/`N`, which would probably be worth adding to the mapping list if "yes"/"no" is). These won't be touched by this new functionality, because we derived the metadata from the source database.

Reply:
Ok, ill remove yes/no values for the this pr then.